virl2_client/event_handling.py

```diff
@@ -66,60 +101,112 @@ def _parse_lab(self, event: Event) -> None:
         elif event.subtype == "state":
             self._parse_lab_state(event)
         else:
-            _LOGGER.warning("Received an invalid lab event (%s)", event.subtype)
+            _LOGGER.warning(f"Received an invalid lab event ({event.subtype})")
 
     @abstractmethod
     def _parse_lab_created(self, event: Event) -> None:
+        """
+        Handle lab created events.
+
+        :param event: An Event object representing the lab created event.
+        """
         pass
 
     @abstractmethod
     def _parse_lab_modified(self, event: Event) -> None:
+        """
+        Handle lab modified events.
+
+        :param event: An Event object representing the lab modified event.
+        """
         pass
 
     @abstractmethod
     def _parse_lab_deleted(self, event: Event) -> None:
+        """
+        Handle lab deleted events.
+
+        :param event: An Event object representing the lab deleted event.
+        """
         pass
 
     @abstractmethod
     def _parse_lab_state(self, event: Event) -> None:
-        pass  # TODO: implement lab state sync via events
+        """
+        Handle lab state events.
+
+        :param event: An Event object representing the lab state event.
+        """
+        pass
 
     def _parse_element(self, event: Event) -> None:
+        """
+        Handle lab element events.
+
+        :param event: An Event object representing the lab element event.
+        """
         if event.subtype == "created":
             self._parse_element_created(event)
         elif event.subtype == "modified":
             self._parse_element_modified(event)
         elif event.subtype == "deleted":
             self._parse_element_deleted(event)
         else:
-            _LOGGER.warning("Received an invalid element event (%s)", event.subtype)
+            _LOGGER.warning(f"Received an invalid element event ({event.subtype})")
```

Comment:
unknown subtype rather than invalid. There's also "state" for state_changed. And the server event_handling.py has these three in all-caps (I guess you lowercase them somewhere).

Reply:
These are for event type "lab_element_event", while state changes are event type "state_change", subtype "state" (idk why).
Capitalization is useless, so it is discarded in Event initialization.